pre_commit/commands/run.py

```diff
@@ -37,7 +43,15 @@ def get_changed_files(new, old):
     )[1].splitlines()
 
 
-def get_filenames(args, include_expr, exclude_expr):
+def filter_filenames_by_types(filenames, types):
+    types = frozenset(types)
+    return tuple(
+        filename for filename in filenames
+        if tags_from_path(filename) >= types
```

Comment:
shrugs, easier to leave it how it is so I'll do that.  If someone has a legitimate usecase for `OR` that neither `exclude_types` nor separate hooks satisfies we can revisit this.